subprojects/base-services/src/test/groovy/org/gradle/internal/service/DefaultServiceRegistryTest.groovy

```diff
@@ -983,13 +978,14 @@ class DefaultServiceRegistryTest extends Specification {
 
         given:
         registry.addProvider(new Object() {
+            ClosableService createService3() {
```

Comment:
This test is supposed to verify ordering, but some of the services already were in the expected close order. So even when I removed the order-preserving code, some of the assertions would pass. This is now fixed and all assertions will fail.